plugins/inputs/elasticsearch_query/elasticsearch_query_test.go

```diff
@@ -484,6 +484,38 @@ var testEsAggregationData = []esAggregationQueryTest{
 		false,
 		false,
 	},
+	{
+		"query 14 - custom date/time format",
+		esAggregation{
+			Index:           testindex,
+			MeasurementName: "measurement14",
+			MetricFields:    []string{"size"},
+			FilterQuery:     "downloads",
+			MetricFunction:  "max",
+			DateField:       "@timestamp",
+			DateFieldFormat: "yyyy",
+			QueryPeriod:     queryPeriod,
+			Tags:            []string{},
+			mapMetricFields: map[string]string{"size": "long"},
+		},
+		[]aggregationQueryData{
+			{
+				aggKey:   aggKey{measurement: "measurement14", name: "size_max", function: "max", field: "size"},
+				isParent: true,
+			},
+		},
+		[]telegraf.Metric{
+			testutil.MustMetric(
+				"measurement14",
+				map[string]string{},
+				map[string]interface{}{"size_max": float64(3318)},
+				time.Date(2018, 6, 14, 5, 51, 53, 266176036, time.UTC),
```

Comment:
This date is the same as the other entries in the table test. With "yyyy" I would have expected a date with Jan 1 at midnight or something.

The tests that use this table are skipped in short mode because they require elasticsearch to run against. Were you able to set up elasticsearch to run them?